### PR TITLE
[GEOT-6536] : BBOX Filter without geom attribute when using FESConfig…

### PR DESCRIPTION
**Pull request by eropartz**
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    encodedFilter.contains(
+                            "<gml:Envelope srsDimension=\"2\" srsName=\"urn:ogc:def:crs:EPSG::4326\">"));
+            Assert.assertTrue(
+                    encodedFilter.contains(
+                            "<gml:lowerCorner>48.817522 -3.475684</gml:lowerCorner>"));
+            Assert.assertTrue(
+                    encodedFilter.contains(
+                            "<gml:upperCorner>48.819155 -3.472614</gml:upperCorner>"));
+
+        } catch (MismatchedDimensionException e) {
```
Comment (aaime) [inline modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java:50]: Letting it simply throw the exception is an easier way to get the test to fail, also gives all details including a stack trace, while this only reports the error message.

File: modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            assertTrue(bbox.getExpression2() instanceof Literal);
+            assertTrue(bbox.getExpression2().evaluate(null) instanceof ReferencedEnvelope);
+
+            ReferencedEnvelope env = (ReferencedEnvelope) bbox.getExpression2().evaluate(null);
+            assertEquals(Double.valueOf(48.817522), env.getMinX());
+            assertEquals(Double.valueOf(-3.475684), env.getMinY());
+            assertEquals(Double.valueOf(48.819155), env.getMaxX());
+            assertEquals(Double.valueOf(-3.472614), env.getMaxY());
+
+        } catch (Exception e) {
```
Comment (aaime) [inline modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java:94]: Same as above.

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BBOXTypeBinding.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public Class getType() {
     public Object getProperty(Object object, QName name) throws Exception {
         BBOX box = (BBOX) object;
 
-        if (FES.ValueReference.equals(name)) {
+        if (FES.expression.equals(name)) {
```
Comment (aaime) [inline modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BBOXTypeBinding.java:85]: Interesting... but legit it seems, from the schema:

```
<xsd:complexType name="BBOXType">
<xsd:complexContent>
<xsd:extension base="fes:SpatialOpsType">
<xsd:choice maxOccurs="2">
<xsd:element ref="fes:expression"/>
<xsd:any namespace="##other"/>
</xsd:choice>
</xsd:extension>
</xsd:complexContent>
</xsd:complexType>
```
Guess the original writer just migrated the code from the superclass, without noticing the schema definition is quite different compared to Filter 1.1, e.g.:

```
<xsd:complexType name="BBOXType">
<xsd:complexContent>
<xsd:extension base="ogc:SpatialOpsType">
<xsd:sequence>
<xsd:element ref="ogc:PropertyName" minOccurs="0"/>
<xsd:element ref="gml:Envelope"/>
</xsd:sequence>
</xsd:extension>
</xsd:complexContent>
</xsd:complexType>
```

File: modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java (hunk 1 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java]: OK, Unit test catch exception removed

File: modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java (hunk 2 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java]: OK, Unit test catch exception removed

**Comment by aaime** [comment]
Thanks, squash-merged the changes. Feel free to open backport PRs for 23.x and 22.x if you wish to.

**Pull request merged**